packages/peregrine/lib/talons/MegaMenu/useSubMenu.js

```diff
@@ -0,0 +1,26 @@
+import { useMemo } from 'react';
+import { useKeyboard } from 'react-aria';
+
+export const useSubMenu = props => {
+    const { isFocused, subMenuState, handleCloseSubMenu } = props;
+
+    const { keyboardProps } = useKeyboard({
+        onKeyDown: e => {
+            //checking for Tab without Shift
+            if (e.keyCode == 9 && !e.shiftKey) {
+                e.target.addEventListener('blur', handleCloseSubMenu());
+            }
+        }
+    });
+
+    const isSubMenuActive = useMemo(() => {
+        if (isFocused) {
+            if (subMenuState) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
```suggestion
       return isFocused && subMenuState;
```

![Screenshot from 2021-08-20 11-28-11](https://user-images.githubusercontent.com/5289370/130212668-586a68c4-a0e4-4641-bfd0-5002ded07987.png)